sql/2020/17_JAMstack/cdn_core_web_vitals_distribution.sql

```diff
@@ -0,0 +1,64 @@
+#standardSQL
+# Core Web Vitals distribution by CDN
+#
+# Note that this is an unweighted average of all sites per SSG.
+# Performance of sites with millions of visitors as weighted the same as small sites.
+SELECT
+  client,
+  CDN,
+  COUNT(DISTINCT origin) AS origins,
+  SUM(fast_lcp) / (SUM(fast_lcp) + SUM(avg_lcp) + SUM(slow_lcp)) AS good_lcp,
+  SUM(avg_lcp) / (SUM(fast_lcp) + SUM(avg_lcp) + SUM(slow_lcp)) AS ni_lcp,
+  SUM(slow_lcp) / (SUM(fast_lcp) + SUM(avg_lcp) + SUM(slow_lcp)) AS poor_lcp,
+
+  SUM(fast_fid) / (SUM(fast_fid) + SUM(avg_fid) + SUM(slow_fid)) AS good_fid,
+  SUM(avg_fid) / (SUM(fast_fid) + SUM(avg_fid) + SUM(slow_fid)) AS ni_fid,
+  SUM(slow_fid) / (SUM(fast_fid) + SUM(avg_fid) + SUM(slow_fid)) AS poor_fid,
+
+  SUM(small_cls) / (SUM(small_cls) + SUM(medium_cls) + SUM(large_cls)) AS good_cls,
+  SUM(medium_cls) / (SUM(small_cls) + SUM(medium_cls) + SUM(large_cls)) AS ni_cls,
+  SUM(large_cls) / (SUM(small_cls) + SUM(medium_cls) + SUM(large_cls)) AS poor_cls,
+FROM
+  `chrome-ux-report.materialized.device_summary`
+JOIN (
+    SELECT
+      CASE
+        WHEN REGEXP_EXTRACT(LOWER(CONCAT(respOtherHeaders, resp_x_powered_by, resp_via, resp_server)), '(x-github-request)') = 'x-github-request' THEN 'GitHub'
+        WHEN REGEXP_EXTRACT(LOWER(CONCAT(respOtherHeaders, resp_x_powered_by, resp_via, resp_server)), '(netlify)') = 'netlify' THEN 'Netlify'
+        WHEN _cdn_provider = 'Microsoft Azure' THEN 'Azure'
+        WHEN _cdn_provider = 'Vercel' THEN 'Vercel'
+        WHEN _cdn_provider = 'Amazon CloudFront' THEN 'AWS'
+        WHEN _cdn_provider = 'Akamai' THEN 'Akamai'
+        WHEN _cdn_provider = 'Cloudflare' THEN 'Cloudflare'
+        ELSE NULL
+      END AS CDN,
+      _TABLE_SUFFIX as client,
+      url,
+      app
+    FROM
+      `httparchive.summary_requests.2020_08_01_*`
```

Comment:
(Throughout) Do you also need a `WHERE firstHtml` clause here so that you're only looking at requests for pages that are served with these headers, not any misc 3P requests?

Reply:
👍

Reply:
Nit: can you take a look at the whitespace related to this change? `firstHtml` is appearing misaligned with the rest of the query.

Reply:
Should be aligned